src/basic/string-util.c

```diff
@@ -317,14 +317,33 @@ char *truncate_nl(char *s) {
         return s;
 }
 
+char ascii_tolower(char x) {
+
+        if (x >= 'A' && x <= 'Z')
+                return x - 'A' + 'a';
+
+        return x;
+}
+
 char *ascii_strlower(char *t) {
         char *p;
 
         assert(t);
 
         for (p = t; *p; p++)
-                if (*p >= 'A' && *p <= 'Z')
-                        *p = *p - 'A' + 'a';
+                *p = ascii_tolower(*p);
+
+        return t;
+}
+
+char *ascii_strlower_n(char *t, size_t n) {
+        size_t i;
+
+        if (n <= 0)
```

Comment:
Yes, it is. But we do that at various cases. I guess it's an excercise in writing code that is robust regarding changing types... Because this way the code will be safe should somebody change the parameter to "int" one day, and forget to update the condition here...

Not that this would be realistic, but it's a way of robust hacking I guess, following the logic of prohibiting everything that is not explicitly OK.